homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -139,7 +139,7 @@ def _get_forecast_from_weather_response(self, weather_response):
 
     def _convert_forecast(self, entry):
         forecast = {
-            ATTR_FORECAST_TIME: entry.reference_time("unix"),
+            ATTR_FORECAST_TIME: dt.utc_from_timestamp(entry.reference_time("unix")),
```

Comment:
Directly convert the incoming forecast Unix timestamps into ISO format. That solves it both for the forecast itself and the derived sensor.